action/protocol/staking/protocol.go

```diff
@@ -458,7 +458,8 @@ func (p *Protocol) Validate(ctx context.Context, act action.Action, sr protocol.
 
 // ActiveCandidates returns all active candidates in candidate center
 func (p *Protocol) ActiveCandidates(ctx context.Context, sr protocol.StateReader, height uint64) (state.CandidateList, error) {
-	c, err := GetStakingStateReader(sr)
+	featureCtx := protocol.MustGetFeatureWithHeightCtx(ctx)
+	c, err := GetStakingStateReader(sr, featureCtx.FixCandCenterAlias(height))
```

Comment:
it was finally called in https://github.com/iotexproject/iotex-core/blob/master/action/protocol/poll/util.go#L88
```
ds, err := p.CalculateCandidatesByHeight(ctx, sr, blkCtx.BlockHeight)
```
so the caller pass in the `height`

of course,  here`blkCtx.BlockHeight` is effectively equal to `sr.Height()`, so we could actually remove `height` from the input parameters, that can be an improvement/refactor later

Reply:
https://github.com/iotexproject/iotex-core/blob/ba21de20289f770af8c2a33fb9f6cd3b3e04c003/action/protocol/poll/util.go#L132

Reply:
i still think we should use `height` here? to give an example, assuming:
```
HF height = 10000
sr.Height() = 10100
epochHeight = 9900
```
in this case, the func tries to read candidates at height 9900, which is before the HF. so we should not enable flags for HF=10000, is that right? 
or maybe I still miss sth, let me know if I'm wrong

Reply:
agree with changing from `height` to `sr.height`, because state from `GetStakingStateReader(sr)` is the state at current chain height.

Reply:
@CoderZhi @Liuhaai  how do you think of the example I give?

Reply:
i'm changing to set the flag = `true`, since it won't impact the results

but I still not convinced whether to use `sr.Height()` or `height`, so write down the below for record

say HF=10000, at this height, the bug will be fixed, with `newCandidateCenter(fixAlias = true)`. Hence for any read request with height < 10000, it should have`fixAlias = false`, and for any read request with height >= 10000, it should have `fixAlias = true`

in other words, for calls like:
https://github.com/iotexproject/iotex-core/blob/ba21de20289f770af8c2a33fb9f6cd3b3e04c003/action/protocol/poll/util.go#L132

input `epochHeight` is useless, it is overriden by `sr.Height()`, which defeats the intention of this function to calculate candidates at `epochHeight`. of course, this limitation is (in another way) due to we don't support archive mode for now, so `sr.Height()` is always the current height, but cannot be set to an earlier height